### PR TITLE
Add API to get data mapper type model

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
$subject
Initially we only consider initializer of variable and function arguments

### CODE REVIEW

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/DataMapper.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                if (optParamName.isPresent()) {
+                    if (optParamName.get().equals(propertyKey)) {
+                        Type t = Type.fromSemanticSymbol(paramSymbol);
+                        if (t != null) {
+                            return gson.toJsonTree(t);
+                        }
+                    }
+                }
+            }
+            throw new IllegalStateException("Type cannot be found for : " + propertyKey);
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/DataMapper.java:88]:
```suggestion
            Optional<Type> type = optParams.flatMap(params -> params.parallelStream()
                    .filter(param -> param.nameEquals(propertyKey)).findAny()).map(Type::fromSemanticSymbol);
            if (type.isEmpty()) {
                throw new IllegalStateException("Type cannot be found for : " + propertyKey);
            }
            return gson.toJsonTree(type.get());
```

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/DataMapper.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        } else {
+            return new JsonObject();
+        }
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/DataMapper.java:91]:
```suggestion
        } 
        return new JsonObject();
```

File: flow-model-generator/modules/flow-model-generator-ls-extension/src/test/java/io/ballerina/flowmodelgenerator/extension/DataMapperTypesTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+        DataMapperTypesRequest request =
+                new DataMapperTypesRequest(sourceDir.resolve(testConfig.source()).toAbsolutePath().toString(),
+                        testConfig.diagram(), testConfig.propertyKey());
+        JsonObject type = getResponse(request).getAsJsonObject("type");
+
+        if (!type.equals(testConfig.type())) {
+            TestConfig updateConfig = new TestConfig(testConfig.source(), testConfig.description(),
+                    testConfig.diagram(), testConfig.propertyKey(), type);
+            updateConfig(configJsonPath, updateConfig);
```
Comment (nipunayf) [inline flow-model-generator/modules/flow-model-generator-ls-extension/src/test/java/io/ballerina/flowmodelgenerator/extension/DataMapperTypesTest.java:61]:
```suggestion
//            updateConfig(configJsonPath, updateConfig);
```

File: flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/DataMapper.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Generates types of the data mapper model.
+ *
+ * @since 1.4.0
+ */
+public class DataMapper {
```
[the remark this answers is not in the document]
Comment (KavinduZoysa) [reply flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/DataMapper.java]: Changed

**Review by nipunayf — APPROVED** [review]

**Pull request merged**